forgescan_realsense/src/scan_image.cpp

```diff
@@ -10,26 +11,48 @@
 #include "rclcpp/rclcpp.hpp"
 #include "std_srvs/srv/empty.hpp"
 #include "geometry_msgs/msg/pose.hpp"
+#include <librealsense2/rs.hpp>
 
 #include "forgescan_realsense/srv/camera_pose.hpp"
+#include "forgescan_realsense/msg/eigen_vector.hpp"
+#include "forgescan_realsense/srv/intrinsics.hpp"
 
 class ScanImage : public rclcpp::Node
 {
     public:
         ScanImage()
         : Node("scan_image")
         {
+            
             realsense_subscriber = this->create_subscription<sensor_msgs::msg::Image>
             ("/camera/forgescan_realsense/camera_image", 10, std::bind(&ScanImage::realsense_image_callback, this, std::placeholders::_1));
             camera_capture_service = this->create_service<forgescan_realsense::srv::CameraPose>(
                 "camera/forgescan_realsense/camera_capture", std::bind(&ScanImage::take_picture, this, std::placeholders::_1, std::placeholders::_2));
-            auto intr = forge_scan::sensor::Intrinsics::create();
-            camera = forge_scan::sensor::Camera::create(intr, 0.0, 100);
         }
     private:
         void take_picture(const std::shared_ptr<forgescan_realsense::srv::CameraPose::Request> request,
```

Comment:
You add a lot more to this class in commit 38eede334970abbc0917320a0a5dc7d278b12f86, could you break out parts of this function into smaller functions to improve readability?